firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/cobs.txt

```diff
@@ -0,0 +1,131 @@
+    THEN("The decoded buffer is empty")    
+
+  WHEN("The COBS::decode method is called on a buffer containing '0x03' as the only byte")      
+    THEN("The decode_cobs method reports out_of_bounds status")
+    THEN("The decoded buffer is empty")
+     
+  WHEN("The COBS::decode method is called on a buffer that contains these bytes '0x05 0x02 0x03' ")    
+    THEN("The decode_cobs method reports out_of_bounds status")
+    THEN("The decoded buffer is empty")
+
+  WHEN("The COBS::decode method is called on a buffer that contains these bytes '0x05 0x02 0xff' ")    
+    THEN("The decode_cobs method reports out_of_bounds status")
+    THEN("The decoded buffer is empty")
+
+  WHEN("The decoded buffer is too small to hold the decoded data") {
+    THEN("The decode_cobs method reports out_of_bounds status")
+    THEN("The decoded buffer is empty")
+
+  WHEN("The COBS::decode method is called on a buffer containing these bytes '\x01\x01'")
+    THEN("The decode_cobs method reports ok status")
+    THEN("The decoded buffer is as expected '0x00'")
+
+  WHEN("The COBS::decode method is called on a buffer containing these bytes '\x01\x01\x01'") {    
+    THEN("The decode_cobs method reports ok status")
+    THEN("The decoded buffer is as expected '0x00 0x00'")
+
+  WHEN("The COBS::decode method is called on a buffer containing these bytes '\x03\x11\x22\x02\x33'")
+    THEN("The decode_cobs method reports ok status")
+    THEN("The decoded buffer is as expected '0x11 0x22 0x00 0x33'")
+
+  WHEN("The COBS::decode method is called on a buffer containing these bytes '\x05\x11\x22\x33\x44'")
+    THEN("The decode_cobs method reports ok status")
+    THEN("The decoded buffer is as expected '\x11\x22\x33\x44'")            
+
+  WHEN("The COBS::decode method is called on a buffer containing these bytes '\x02x'") {
+    THEN("The decode_cobs method reports ok status")
+    THEN("The decoded buffer is as expected 'x' ")
+
+  WHEN("The COBS::decode method is called on a buffer containing these bytes '\x03xy'")    
+    THEN("The decode_cobs method reports ok status")
+    THEN("The decoded buffer is as expected 'xy' ")
+
+  WHEN("The COBS::decode method is called on a buffer containing the bytestring '\x0cHello World'")
+    THEN("The decode_cobs method reports ok status")
+    THEN("The decoded buffer is as expected 'Hello World'")
+
+  WHEN("The COBS::decode method is called on a buffer containing a encoded sensor measurements message payload")
+    THEN("The decode_cobs method reports ok status")
+    THEN("The decoded buffer is as expected")
+
+  WHEN("The COBS::decode method is called on a buffer that contains these bytes '0x03, 0x9c, 0xb8, 0x03, 0xbe, 0xce'")
+    THEN("The decode_cobs method reports ok status")
+    THEN("The decoded buffer is as expected '\x9c\xb8\x00\xbe\xce' ")
+
+  WHEN("The COBS::decode method is called on a encoded buffer of capacity 254 bytes whose original buffer was of capacity 253 with no null bytes")
```

Comment:
Let's figure out how to state this WHEN and the next two WHENs more clearly and unambiguously.

Reply:
We'll use "is called on a 257-byte buffer filled with the manual encoding of a (e.g.) 255 non-null-byte payload"